src/rust/engine/fs/src/snapshot.rs

```diff
@@ -496,6 +498,36 @@ impl Snapshot {
       })
       .to_boxed()
   }
+
+  pub fn from_files_content(
+    store: Store,
+    files_content: Vec<FileContent>,
+  ) -> impl Future<Item = Snapshot, Error = String> {
+    let files_content: BTreeMap<File, Bytes> = files_content
+      .into_iter()
+      .map(|fc| {
+        let file = File {
+          path: fc.path,
+          // TODO: Configurable?
+          is_executable: true,
+        };
+        (file, fc.content)
+      })
+      .collect();
+    // TODO: Should there be a way to represent empty directories?
```

Comment:
"Yes, when needed".

If we were to represent directories the same way we do on Snapshot, adding directories is cheap to do later.

Adding the executable bit is possibly more awkward to do later though. So maybe that should happen sooner, but I don't have a design in mind.

Reply:
Added `is_executable` to `FileContent` because why not.

Will leave empty directories until needed.

Reply:
Nice! Thanks.